tests/inputs/stmtTests.mg

```diff
@@ -17,4 +17,107 @@ implementation StmtTests = {
         // Invalid assignment
         t1 = mkT2();
     }
-};
+}
+
+// TODO: add more tests
+implementation VarExprTests = {
+
+    require type T;
+    require type U;
+
+    procedure conflicting_type_annotation() = {
+        var x: T;
+        // should succeed
+        x: T;
+        // should fail
+        x: U;
+    }
+}
+
+
+// TODO: add more tests
+implementation CallExprTests = {
+
+    require type T;
+    require type U;
+
+    function overload_return_type(): T;
+    function overload_return_type(): U;
+
+    procedure call_functions() = {
```

Comment:
nit: can you think of a better name for this procedure? Maybe something like `deduce_overloaded_type`?